### PR TITLE
Cosmos newsletter form

### PR DESCRIPTION
**Pull request by fadeev**

### CODE REVIEW

**Comment by fadeev** [comment]
@nassdonald this is still WIP, just want to know your opinion. Since "banner" and "page" differ mostly in terms of height, I've added an option to toggle between them. With both height and content transitions it kind of looks good even for a banner.

https://deploy-preview-169--cosmos-ui.netlify.com/iframe.html?id=cosmosnewslettersignup--normal&knob-fullscreen=false
https://deploy-preview-169--cosmos-ui.netlify.com/iframe.html?id=cosmosnewslettersignup--normal&knob-fullscreen=true

**Comment by nassdonald** [comment]
Wonderful stuff!! 👏 

I like the idea. The main reason for using a modal and not just showing subsequent screens in the banner was to do with users not having a good reference for the "top" of the relevant area of the page, since there's content above and below and the height of the banner will change - with user scrolling inevitable.

Though, we will have to programmatically scroll users to to the top of the page after each screen transition anyway, so maybe not a huge deal. We could try it out and see how it feels - maybe get a few people to test it?

**Comment by fadeev** [comment]
> We could try it out and see how it feels - maybe get a few people to test it?

I can finish up the banner-style UI and replace the existing form on staging, so that we can see how it holds up in the context of the rest of the website.

**Comment by fadeev** [comment]
Oh, and btw transition between graphics on the left doesn't work properly because of SVG gradients. The envelope image transitions in with solid shapes and then gradients pop in. I'll see what kind of #000 magic is this tomorrow.

**Comment by nassdonald** [comment]
That'd be good to get to the bottom of actually, because if we can't transition SVG elements with gradients, we might have a problem. 😟

**Comment by fadeev** [comment]
@nassdonald ready for review.

https://www.youtube.com/watch?v=4cMxVajFNo0

**Comment by nassdonald** [comment]
This is looking great! Only thing left to do is to adapt it for smaller screens. I'll mock up something quickly in Figma for this.

I've tweaked the interaction states and fixed one line-height on the confirmation screen.

File: src/NewsletterSignup/IbcNewsletterSignup.vue [hunk 1]
```diff
@@ -0,0 +1,364 @@
...
+        tools: false,
+        ecosystem: false
+      },
+      url: "https://zcs1.maillist-manage.com/campaigns/weboptin.zc",
+      commonFormData: {
+        "zc_trackCode": "ZCFORMVIEW",
+        "viewFrom": "URL_ACTION",
+        "submitType": "optinCustomView",
+        "emailReportId": "",
+        "zx": "129a50c11",
```
Comment (lovincyrus) [inline src/NewsletterSignup/IbcNewsletterSignup.vue:312]: is this zoho's newsletter id for ibc? @fadeev

Comment (fadeev) [reply src/NewsletterSignup/IbcNewsletterSignup.vue:312]: No, it is not. You need `lD`, `zcld` and `zc__formIx`. If the subscription link on `cosmos.network/ibc` is correct, then we can fish out these values by inspecting the source of the form.

```js
{
  lD: "16352f88328053f9",
  zcld: "16352f88328053f9",
  zc_formIx: "4ef47fbb86ab6668b78ce576d7aa791940fbf6cd35f5c50c"
}
```

Please, double check.

Comment (lovincyrus) [reply src/NewsletterSignup/IbcNewsletterSignup.vue:312]: image

**Comment by nassdonald** [comment]
This still hasn't been optimized for smaller devices, so I suggest we fix that before we merge.

image

I'll add mockups in Figma, as promised.

**Pull request merged**